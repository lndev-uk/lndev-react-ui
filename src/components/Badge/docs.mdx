import { Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Badge from './';

# Badge

If Size is not specified the Badges scale to match the size of the immediate parent element by using relative font sizing and em units.

<Preview>
  <Story id="components-badge--example" />
</Preview>
<Props of={Badge} />

## UX/UI Variants

Badges can be use in different variants styles based on de **UX/UI** System

<Preview>
  <Story id="components-badge--variants" />
</Preview>

## Sizing

Overriding Inherint sizing.

<Preview>
  <Story id="components-badge--sizes" />
</Preview>
