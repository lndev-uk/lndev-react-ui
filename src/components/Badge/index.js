//@flow
import React from 'react';
import PropTypes from 'prop-types';

type BadgeType =
  | 'primary'
  | 'secondary'
  | 'dark'
  | 'success'
  | 'warning'
  | 'error';

type BadgeSize = 'big' | 'small';

export type BadgeProps = {
  label: String,
  size: BadgeSize,
  type: BadgeType
};

export function Badge(props: BadgeProps): React.ReactNode {
  const { size, type, label } = props;
  const sizeClassName = size ? 'badge--' + size : '';
  const typeClassName = type ? 'badge--' + type : '';
  return (
    <div className={`badge ${typeClassName} ${sizeClassName}`}>{label}</div>
  );
}

Badge.propTypes = {
  label: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['big', 'small']),
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'dark',
    'success',
    'warning',
    'error'
  ])
};

export default Badge;
