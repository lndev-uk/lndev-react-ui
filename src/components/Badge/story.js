import React from 'react';
import Badge from '.';

export default { title: 'Badge' };

export const Types = () => (
  <div>
    <div>
      <Badge type="primary" label="Primary" />
    </div>
    <div>
      <Badge type="secondary" label="Secondary" />
    </div>
    <div>
      <Badge type="dark" label="Dark" />
    </div>
    <div>
      <Badge type="success" label="Success" />
    </div>
    <div>
      <Badge type="warning" label="Warning" />
    </div>
    <div>
      <Badge type="error" label="Error" />
    </div>
  </div>
);

export const Sizes = () => (
  <div>
    <div>
      <Badge size="small" label="Small" />
    </div>
    <div>
      <Badge label="Default" />
    </div>
    <div>
      <Badge size="big" label="Big" />
    </div>
  </div>
);
