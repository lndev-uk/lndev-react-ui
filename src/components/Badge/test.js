import React from 'react';
import { render } from '@testing-library/react';
import Badge from '.';

describe('Badge', () => {
  it('matches snapshot success', () => {
    const { container } = render(<Badge type="primary" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot secondary', () => {
    const { container } = render(<Badge type="secondary" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot dark', () => {
    const { container } = render(<Badge type="dark" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot success', () => {
    const { container } = render(<Badge type="success" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot warning', () => {
    const { container } = render(<Badge type="warning" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot error', () => {
    const { container } = render(<Badge type="error" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('display correct text passed by label', () => {
    const { getByText } = render(<Badge type="error" label="test label" />);
    expect(getByText('test label').innerHTML).toBe('test label');
  });
});
