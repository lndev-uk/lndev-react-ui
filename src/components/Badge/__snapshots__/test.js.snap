// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge matches snapshot dark 1`] = `
<div>
  <div
    class="badge badge--dark "
  >
    test label
  </div>
</div>
`;

exports[`Badge matches snapshot error 1`] = `
<div>
  <div
    class="badge badge--error "
  >
    test label
  </div>
</div>
`;

exports[`Badge matches snapshot secondary 1`] = `
<div>
  <div
    class="badge badge--secondary "
  >
    test label
  </div>
</div>
`;

exports[`Badge matches snapshot success 1`] = `
<div>
  <div
    class="badge badge--primary "
  >
    test label
  </div>
</div>
`;

exports[`Badge matches snapshot success 2`] = `
<div>
  <div
    class="badge badge--success "
  >
    test label
  </div>
</div>
`;

exports[`Badge matches snapshot warning 1`] = `
<div>
  <div
    class="badge badge--warning "
  >
    test label
  </div>
</div>
`;
